packages/devtools_app/lib/src/screens/deep_link_validation/deep_link_list_view.dart

```diff
@@ -354,3 +331,83 @@ class _DomainCheckTable extends StatelessWidget {
     );
   }
 }
+
+class _DeepLinkListViewTopPanel extends StatefulWidget {
+  const _DeepLinkListViewTopPanel();
+
+  @override
+  State<_DeepLinkListViewTopPanel> createState() =>
+      _DeepLinkListViewTopPanelState();
+}
+
+class _DeepLinkListViewTopPanelState extends State<_DeepLinkListViewTopPanel>
+    with
+        ProvidedControllerMixin<DeepLinksController,
+            _DeepLinkListViewTopPanel> {
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    initController();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.end,
+      children: [
+        Padding(
+          padding: const EdgeInsets.all(defaultSpacing),
+          child: ValueListenableBuilder(
+            valueListenable: controller.selectedVariantIndex,
+            builder: (_, value, __) {
+              return _AndroidVariantDropdown(
+                androidVariants:
+                    controller.selectedProject.value!.androidVariants,
+                index: value,
+                onVariantIndexSelected: (index) {
+                  controller.selectedVariantIndex.value = index;
+                },
+              );
+            },
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class _AndroidVariantDropdown extends StatelessWidget {
+  const _AndroidVariantDropdown({
+    required this.androidVariants,
+    required this.index,
+    required this.onVariantIndexSelected,
+  });
+
+  final List<String> androidVariants;
+  final int index;
+  final ValueChanged<int> onVariantIndexSelected;
+  @override
+  Widget build(BuildContext context) {
+    final List<DropdownMenuItem<int>> items = <DropdownMenuItem<int>>[];
+    for (int i = 0; i < androidVariants.length; i += 1) {
+      items.add(
+        DropdownMenuItem<int>(
+          value: i,
+          child: Text(androidVariants[i]),
+        ),
+      );
+    }
+    return Row(
+      children: [
+        const Text('Android Variant:'),
+        RoundedDropDownButton<int>(
+          value: index,
+          items: items,
```

Comment:
nit this can be even more terse by defining the list here:
```dart
items: [
  for (final int i = 0; i < androidVariants.length; i++)
    DropdownMenuItem<int>(
      value: i,
      child: Text(androidVariants[i]),
    ),
],
```